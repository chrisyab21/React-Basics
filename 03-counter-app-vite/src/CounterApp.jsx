
import PropType from 'prop-types'
import {useState} from 'react';

const EventoAfuera = (event, newvalue)=>{
    console.log(event);
    console.log(newvalue);
    
}


export const CounterApp = ({value, subtitle})=>{

    
    const inicial = value;

    const [counter, setCounter ] = useState(value)

    const ButtonSumar = (event)=>{
       // console.log(event);
        
        setCounter(counter+1);
    }

    
    const ButtonRestar = ()=>{
        
        setCounter((c)=> c-1);
        // setCounter(counter-1);
    }


    const ButtonReset = ()=>{
       
        setCounter(()=> inicial);
    }

    return (
        <>
        
        <h1>CounterApp</h1>

        <h2>{value}</h2>
        <h2>{subtitle}</h2>
        <h2>Contador: {counter}</h2>
        <button onClick={ButtonSumar}>
            +1
        </button>
        <button onClick={(event)=> ButtonRestar(event)}>
            -1
        </button>
        <button aria-label="btn-reset" onClick={()=> ButtonReset()}>
            Reset
        </button>
        </>
    );
}


CounterApp.propTypes = {

    value: PropType.number.isRequired
}


CounterApp.defaultProps = {

    value: 21
}


