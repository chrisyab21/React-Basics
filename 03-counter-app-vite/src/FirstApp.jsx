
import PropTypes from 'prop-types';


export const FirstApp= ({title, subtitle, subtitle2}) =>{

    return(  
    <>
        <h1 data-testid="test-title">{title} </h1>
        <p>{subtitle}</p>
        <p>{subtitle2+1}</p>
    </>
    );
}


FirstApp.propTypes ={

    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired,
    subtitle2: PropTypes.number.isRequired
}


FirstApp.defaultProps ={

    title: "Hola soy Goku",    
    subtitle: "Kimi Monogatari...",
    subtitle2: 455
}