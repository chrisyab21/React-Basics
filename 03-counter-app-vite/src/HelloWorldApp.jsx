

export const App =  ()=>{

    return <div>Hola Mundo!</div>;
}



