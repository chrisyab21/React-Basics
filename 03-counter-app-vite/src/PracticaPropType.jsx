import PropTypes from 'prop-types';

//Utilizando las Props
export const PropsComponent= ({title = 'Hola Fans de Naruto', subtitle = '99' }) =>{

    return(  
    <>
        <h1>{title} </h1>
        <p>{subtitle+1}</p>
    </>
    );
}



//Utilizando las Props pero ahora aplicando PropTypes
export const PropTypeComponent= ({title, subtitle}) =>{

    return(  
    <>
        <h1>{title} </h1>
        <p>{subtitle+1}</p>
    </>
    );
}


PropTypeComponent.propTypes ={

    title: PropTypes.string.isRequired,
    subtitle: PropTypes.number.isRequired
}


PropTypeComponent.defaultProps ={

    title: "No hay titulo",
    subtitle: 455
}