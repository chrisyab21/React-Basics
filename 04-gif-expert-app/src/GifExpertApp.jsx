
import { useState } from 'react';


import React from 'react'

import { AddCategory, GifGrid } from './Componentes/imports';
//import { GifGrid } from './Componentes/GifGrid';


export const GifExpertApp = () => {


  const [categorias, setCategory] = useState(['One Piece']);

 // console.log(categorias);

  const agregarCategoria = (newCategory) =>{

  var converted = newCategory.toLowerCase();

    if(categorias.includes(converted)) return;

    setCategory([converted,...categorias]);

    //setCategory((cat) => [...cat,'Vice City']);     
  }



  return (
    <>
        <h1>GifExpertApp</h1>


        <AddCategory onAddCategory={(value) => agregarCategoria(value)}></AddCategory>


        {/* <AddCategory onAddCategory={setCategory}></AddCategory> */}

        
     
          {       
          categorias.map((categoria)=>{ 
            
            
              return (
             
                <GifGrid key={categoria} category={categoria}></GifGrid>
              )
            })          
          }
    
    </>
  )
}




