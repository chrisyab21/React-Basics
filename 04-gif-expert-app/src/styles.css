* {
  font-family: Helvetica, Arial, sans-serif;
  background-color: rgb(234, 234, 234);
}

body {
  padding: 60px;
}

input {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgb(97, 32, 158);
  color: black;
  font-size: 1.2rem;
  outline: none;
  padding: 10px 15px;
  width: 100%;
}

h2 {
  font-size: 1.5rem;
}

h3 {
  font-size: 3rem;
  margin-bottom: 5px;
}

.card-grid{
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-between;
}


.card {
  align-content: center;
  align-items: center;
  background-color: black;
  border-radius: 10px;
  box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
  display: flex;
  flex-direction: column;
  height: 0%;  
  justify-content: center;
  padding-bottom: 20px;
  padding-top: 20px;
  padding-right: 20px;
  padding-left: 20px;
  overflow: hidden;
  margin-bottom: 20px;
  flex-basis: 30%;
  

}



.card p {
  background-color: goldenrod;
  flex: 1;
  font-size: 1.5rem;
  margin-top: 10px;
  padding: 5px 20px 0px 20px;
  text-align: center;
  color: #000;
  border: 0.5px solid gold ;
  width: auto;
  margin-bottom: 0px;


}

.card img{
  width: 500px;
  height: 250px;
}

div li{

  /* background-color: aqua; */
  height: 20px;
}