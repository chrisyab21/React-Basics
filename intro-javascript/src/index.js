
const activo = true;


let mensaje ='';

if(activo){

  mensaje = 'Activo'

}else{

mensaje='Inactivo'

}
console.log(mensaje);


//Operador Ternario
const mensaje2 = (activo) ? 'Activo' : 'Inactivo'

console.log(mensaje2);

//Operador ternario pero que solo devuelve algo cuando es true, usando el operador logico AND(&&)
//Si la variable activo es TRUE entonces devuelve el texto Activo, pero si la primera condicion da false
//entonces devuelve un false y no evaluda la segunda condicion. Porque el operador AND(&&) devuelve un false
//cuando cualquiera de sus condiciones da false.
const mensaje3 = (activo) && 'Activo';

console.log(mensaje3);




const getImage = async()=>{


  try {
    
    const apikey = 'HVoHbv8C8iXNnCl8ypKS6RKZHa4u49XM';


    const respuesta = await fetch(`https://api.giphy.com/v1/gifs/random?api_key=${apikey}`);
    
    const {data} = await respuesta.json();
    
    
    const {url} = data.images.original
    console.log(url);
    
    const img =  document.createElement('img');
    
    img.src = url;
    
    document.body.append(img);
    

  } catch (error) {

    console.error('Hubo un error', error);
    
  }

}

getImage();
