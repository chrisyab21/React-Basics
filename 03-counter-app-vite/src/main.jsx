import React, { StrictMode } from "react";

import ReactDOM from "react-dom/client";

import {App} from './HelloWorldApp';
import {FragmentoApp, FragmentoApp2,VariablesPractica, RetornoFunciones} from './Practicas'
 import { FirstApp } from "./FirstApp";


import { PropTypeComponent , PropsComponent} from "./PracticaPropType";
import { CounterApp } from "./CounterApp";
import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(

    <React.StrictMode>
        {/* <FirstApp></FirstApp>
       <FragmentoApp></FragmentoApp>
       <FragmentoApp2></FragmentoApp2>
       <VariablesPractica></VariablesPractica>
       <RetornoFunciones></RetornoFunciones>
       <PropsComponent title="Hola soy Beyonder" subtitle={123}></PropsComponent>
       <PropTypeComponent ></PropTypeComponent>  */}
       <CounterApp value={2016} subtitle='Christian'></CounterApp> 
    </React.StrictMode>
);