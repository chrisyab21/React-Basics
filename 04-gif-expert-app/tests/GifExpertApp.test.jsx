import { fireEvent, render, screen } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";


describe('Pruebas en el component GifExpertApp', () => { 

    test('Verificar que si añada una categoria y luego limpie el input', () => { 


            const category = 'Dragon Ball Gt'

           const {container} = render(<GifExpertApp></GifExpertApp>)


            expect(container).toMatchSnapshot();


            const input = screen.getByRole('textbox');

            const form = screen.getByRole('form');

         

            fireEvent.input(input, { target: { value: category}});

            expect(input.value).toBe(category);

            fireEvent.submit(form);
           
            expect(screen.getAllByRole("heading", {level:3}).length).toBe(2);
            expect(input.value).toBe('');

            screen.debug();

     });


        test('Verificar que si el input esta en blanco NO añada una categoria', () => { 


            const category = ''

            render(<GifExpertApp></GifExpertApp>)

            const input = screen.getByRole('textbox');
            const form = screen.getByRole('form'); 

            fireEvent.input(input, { target: { value: category}});

            fireEvent.submit(form);

            expect(screen.getAllByRole("heading", {level:3}).length).toBe(1);

            screen.debug();

    });



    
    test('Verificar que No añada categorias repetidas', () => { 


        const category = 'Dragon Ball'

        render(<GifExpertApp></GifExpertApp>)

        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form'); 

        fireEvent.input(input, { target: { value: category}});

        fireEvent.submit(form);

        fireEvent.input(input, { target: { value: category}});

        fireEvent.submit(form);

        expect(screen.getAllByRole("heading", {level:3}).length).toBe(2);

        screen.debug();

});


 });