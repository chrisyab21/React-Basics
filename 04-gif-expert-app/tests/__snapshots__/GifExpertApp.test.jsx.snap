// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el component GifExpertApp Verificar que si a√±ada una categoria y luego limpie el input 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    One Piece
  </h3>
  <h2>
    Cargando...
  </h2>
  <ol
    class="card-grid"
  />
</div>
`;
